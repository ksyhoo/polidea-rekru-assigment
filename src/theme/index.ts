export interface Theme {
  borderRadius: string;
  colors: {
    white: string;
    black: string;
    gray: string;
    main: string;
    secondary: string;
    accent: string;
  };
  spacing: {
    small: string;
    medium: string;
    large: string;
  };
  font: string;
}

const theme: Theme = {
  borderRadius: '8px',
  colors: {
    white: '#fff',
    black: '#000',
    gray: '#f5f5f5',
    main: '#d6d6d6',
    secondary: '#616161',
    accent: '#e38212',
  },
  spacing: {
    small: '8px',
    medium: '16px',
    large: '32px',
  },
  font: 'Poppins',
};

export default theme;
