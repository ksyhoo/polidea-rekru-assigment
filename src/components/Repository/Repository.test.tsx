import { create } from 'react-test-renderer';
import React from 'react';
import Repo from '.';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import theme from 'theme';
import { BrowserRouter } from 'react-router-dom';

describe('RepoComponent', () => {
  it('render Repo component', async () => {
    const wrapper = create(
      <BrowserRouter>
        <ThemeProvider theme={theme}>
          <Repo
            repo={{
              name: 'repo',
              owner: { name: ' owner', type: 'any' },
              numberOfStars: 22,
              primaryLanguage: 'Ponglish',
              id: 1,
            }}
          />
        </ThemeProvider>
      </BrowserRouter>
    ).toJSON();

    await Repo;
    expect(wrapper).toMatchSnapshot();
  });
});
