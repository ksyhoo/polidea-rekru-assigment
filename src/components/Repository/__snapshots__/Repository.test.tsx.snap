// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoComponent render Repo component 1`] = `
.c2 {
  font-size: 18px;
  color: #000;
  margin: 3px;
  margin-bottom: 12px;
  font-weight: normal;
}

.c2 span {
  color: #616161;
}

.c4 {
  font-size: 12px;
  color: #000;
  margin: 3px;
  margin-bottom: undefinedpx;
  font-weight: normal;
}

.c4 span {
  color: #616161;
}

.c6 {
  font-size: 6px;
  color: #000;
  margin: 3px;
  margin-bottom: undefinedpx;
  font-weight: normal;
}

.c6 span {
  color: #616161;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 29px 28px 195px -53px rgba(0,0,0,1);
  padding: 32px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c5 {
  background: url('star.png') no-repeat center center;
  background-size: contain;
  height: 70px;
  width: 70px;
}

.c5 .c1 {
  color: #fff;
  font-size: 24px;
  line-height: 70px;
  text-align: center;
  text-shadow: 2px 2px #e38212;
}

.c3:before {
  content: '';
  border: 2px solid #e38212;
  margin-right: 5px;
}

<a
  className="c0"
  href="/repository/repo/ owner"
  onClick={[Function]}
>
  <p
    className="c1 c2 c3"
    fontSize={18}
  >
    repo
  </p>
  <p
    className="c1 c4"
    fontSize={12}
  >
     owner
     / 
    <span>
      any
    </span>
  </p>
  <p
    className="c1 c4"
    fontSize={12}
  >
    Ponglish
  </p>
  <div
    className="c5"
  >
    <p
      className="c1 c6"
      fontSize={6}
    >
      22
    </p>
  </div>
</a>
`;
